packages/patternfly-3/react-console/src/VncConsole/VncActions.js

```diff
@@ -1,30 +1,60 @@
 import React from 'react';
+import ReactDOM from 'react-dom';
 import PropTypes from 'prop-types';
 import { MenuItem, Button, helpers } from 'patternfly-react';
 
 const { Dropdown } = Button;
 const { noop } = helpers;
 
-const VncActions = ({ textSendShortcut, textCtrlAltDel, onCtrlAltDel }) => (
-  <Dropdown bsStyle="default" title={textSendShortcut} id="console-send-shortcut" onClick={noop}>
-    <MenuItem eventKey="1" onClick={onCtrlAltDel}>
-      {textCtrlAltDel}
-    </MenuItem>
-  </Dropdown>
-);
+const VncActions = ({
+  textSendShortcut,
+  textCtrlAltDel,
+  textDisconnect,
+  onCtrlAltDel,
+  onDisconnect,
+  portalToolbarTo
+}) => {
+  const toolbar = (
+    <div>
+      <Dropdown bsStyle="default" title={textSendShortcut} id="console-send-shortcut" onClick={noop}>
+        <MenuItem eventKey="1" onClick={onCtrlAltDel}>
+          {textCtrlAltDel}
+        </MenuItem>
+      </Dropdown>
+      <Button bsStyle="default" onClick={onDisconnect}>
+        {textDisconnect}
+      </Button>
+    </div>
+  );
+
+  if (!portalToolbarTo) {
+    return toolbar;
+  }
+  return (
+    document.getElementById(portalToolbarTo) && ReactDOM.createPortal(toolbar, document.getElementById(portalToolbarTo))
+  );
+};
 
 VncActions.propTypes = {
   onCtrlAltDel: PropTypes.func,
+  onDisconnect: PropTypes.func,
 
   textCtrlAltDel: PropTypes.string,
-  textSendShortcut: PropTypes.string
+  textSendShortcut: PropTypes.string,
+  textDisconnect: PropTypes.string,
+
+  portalToolbarTo: PropTypes.string
 };
 
 VncActions.defaultProps = {
   onCtrlAltDel: noop,
+  onDisconnect: noop,
```

Comment:
This is a breaking change.
The consuming application is **silently** required to add implementation. Otherwise the button just appears there without any function.

Reply:
Ok, I'll do `onDisconnect` props required, but this is was just copy from `onCtlAltDel`. So why in this case `onCtrlAltDel: noop` not required too?